plugins/automation/app/services/discourse_automation/destroy.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class DiscourseAutomation::Destroy
+  include Service::Base
+
+  # @!method self.call(guardian:, params:)
+  #   @param [Guardian] guardian
+  #   @param [Hash] params
+  #   @option params [Integer] :automation_id
+  #   @return [Service::Base::Context]
+
+  policy :can_destroy_automation
```

Comment:
I moved the policy to be the first step, as it only relies on `guardian`. So we can stop the execution flow even before validating any param or fetching any model.